public.yaml

```diff
@@ -65,17 +65,18 @@ metadata:
         "image": "sles12/salt-master:2016.11.4",
         "command": ["sh", "-c", "umask 377;
                                  mkdir /salt-master-pki/minions/;
+                                 temp_dir=`mktemp -d`;
                                  if [ ! -f /salt-admin-minion-pki/minion.pub ] || [ ! -f /salt-admin-minion-pki/minion.pem ]; then
-                                   salt-key --gen-keys=admin --gen-keys-dir /tmp;
-                                   cp /tmp/admin.pub /salt-master-pki/minions/admin;
-                                   mv /tmp/admin.pub /salt-admin-minion-pki/minion.pub;
-                                   mv /tmp/admin.pem /salt-admin-minion-pki/minion.pem;
+                                   salt-key -u root --gen-keys=admin --gen-keys-dir $temp_dir;
```

Comment:
I assume the addition of `-u root` is for forcing salt to match the perms of the mktemp created directly?